### PR TITLE
Add config setting for hmac signature algorithm

### PR DESCRIPTION
**Pull request by andrenpaes**
Part of the effort for migrating to the `HmacSHA256` signing algorithm.

Changes are pretty straightforward, but since some things got refactored, it touches a lot of code.

### CODE REVIEW

**Comment by pierre** [comment]
👍

File: src/main/java/org/killbill/billing/plugin/adyen/client/AdyenConfigProperties.java [hunk 1]
```diff
@@ -84,6 +88,14 @@ public AdyenConfigProperties(final Properties properties) {
         this.hmacSecrets = properties.getProperty(PROPERTY_PREFIX + "hmac.secret");
         refillMap(secretMap, hmacSecrets);
 
+        final String hmacAlgorithm = properties.getProperty(PROPERTY_PREFIX + "hmac.algorithm");
+        if (hmacAlgorithm != null) {
```
Comment (sbrossie) [inline src/main/java/org/killbill/billing/plugin/adyen/client/AdyenConfigProperties.java]: Nit: `this.hmacAlgorithms = Objects.firstNonNull(properties.getProperty(PROPERTY_PREFIX + "hmac.algorithm"),  DEFAULT_HMAC_ALGORITHM);`

File: src/main/java/org/killbill/billing/plugin/adyen/client/AdyenConfigProperties.java [hunk 2]
```diff
@@ -176,6 +188,14 @@ public String getHmacSecret(final String countryIsoCode) {
         return secretMap.get(adjustCountryCode(countryIsoCode));
     }
 
+    public String getHmacAlgorithm(final String countryIsoCode) {
+        if (countryIsoCode == null || hmacAlgorithmMap.isEmpty()) {
```
Comment (sbrossie) [inline src/main/java/org/killbill/billing/plugin/adyen/client/AdyenConfigProperties.java]: Do we have to be cautious about the `UK` vs `GB` confusion? I guess whatever Adyen sees is what you we should use inside this plugin.

**Comment by sbrossie** [comment]
👍

**Pull request merged**